### ISSUE TITLE
possibly-missing-attribute warning on Union type narrowing based on Literal values

### ISSUE THREAD

**Issue by arthur-st** [astral-sh/ty#2022]
### Summary

```python
from pydantic import BaseModel
from typing import Literal

class Foo(BaseModel):

    foo: str
    kind: Literal["foo"] = "foo"

class Bar(BaseModel):

    bar: str
    kind: Literal["bar"] = "bar"

FooBar = Foo | Bar

def baz(thing: FooBar):
    if thing.kind == "bar":
        print(thing.bar)
```

This code shows the following warning:

```
❯ ty check ty-test.py
warning[possibly-missing-attribute]: Attribute `bar` may be missing on object of type `Foo | Bar`
  --> ty-test.py:18:15
   |
16 | def baz(thing: FooBar):
17 |     if thing.kind == "bar":
18 |         print(thing.bar)
   |               ^^^^^^^^^
   |
info: rule `possibly-missing-attribute` is enabled by default
```

This warning is not correct, since `Bar` is the only type with the `kind="bar"` attribute.

### Version

ty 0.0.2 (Homebrew 2025-12-16)

**Comment by arthur-st** [comment]
Related #2023

**Comment by AlexWaygood** [comment]
Thanks for the report. Please see #1479
